e2e/validate/validation.go

```diff
@@ -181,3 +186,45 @@ func clustersInClusterSet(env *types.Env, config *types.Config, log *zap.Sugared
 
 	return nil
 }
+
+// detectDistro detects the cluster distribution. If distro is set in config, it uses the user
+// provided distro. Otherwise, it determines the distro, sets config.Distro and config.Namespaces.
+// Returns an error if distro detection fails or if clusters have inconsistent distributions.
+func detectDistro(env *types.Env, cfg *types.Config, log *zap.SugaredLogger) error {
+	if cfg.Distro != "" {
+		log.Infof("Using user selected kubernetes distribution: %q", cfg.Distro)
+
+		return nil
+	}
+
+	isHubOpenShift, err := IsOpenShiftCluster(env.Hub)
+	if err != nil {
+		return fmt.Errorf("failed to detect distro for cluster %q: %w", env.Hub.Name, err)
+	}
+
+	isC1OpenShift, err := IsOpenShiftCluster(env.C1)
+	if err != nil {
+		return fmt.Errorf("failed to detect distro for cluster %q: %w", env.C1.Name, err)
+	}
+
+	isC2OpenShift, err := IsOpenShiftCluster(env.C2)
+	if err != nil {
+		return fmt.Errorf("failed to detect distro for cluster %q: %w", env.C2.Name, err)
+	}
+
+	if isHubOpenShift != isC1OpenShift || isHubOpenShift != isC2OpenShift {
+		return fmt.Errorf("unsupported configuration, clusters have inconsistent distributions")
+	}
+
```

Comment:
This code is repetitive, I think we can do this in a nice way.

First we want to change the helper to return the distro name instead of bool:

   func probeDistro(cluster *types.Cluster) (string, err) {
       ...
   }

Then we can use a loop:

    var detectedDistro string
    for _, cluster := range []*types.Cluster[&env.Hbu, &env.C1, &env.C2} {
        distro, err := probeDistro(cluster)
        if err != nil {
            return fmt.Errorf("failed to determine cluster distro: %s", err)
        }
        if detectedDistro == "" {
            detectedDistro = distro
        } else if detectedDistro != distro {
            return fmt.Errorf("unsupported setup ...")
        }
    }